pkg/logql/engine.go

```diff
@@ -256,7 +256,10 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	statResult := statsCtx.Result(time.Since(start), queueTime, q.resultLength(data))
 	statResult.Log(level.Debug(spLogger))
 
-	status, _ := server.ClientHTTPStatusAndError(err)
+	status := 200
+	if err != nil {
+		status, _ = server.ClientHTTPStatusAndError(err)
```

Comment:
Hmm does it make sense to add the `err != nil` check in `ClientHTTPStatusAndError()` and return 200 there?

Also, this would benefit from a unit test and I'm not sure how easy it would be to do it here (because of the direct call to `RecordRangeAndInstantQueryMetrics()` that I am not sure we can mock), while doing it in the  `ClientHTTPStatusAndError()` seems much easier.

Reply:
yeah, it makes sense to add a check there.